service/serializer/src/main/java/org/jolokia/service/serializer/GSonSerializer.java

```diff
@@ -0,0 +1,201 @@
+                json = JsonPath.parse(json, jsonPathconfiguration).read(path).toString();
+            }
+            return wrapString(json);
+        } catch (PathNotFoundException e) {
+            getJolokiaContext().error(String.format("Unexpected error when serializing %s", pValue.getClass().getSimpleName()), e);
+            return null;
+        } catch (InvalidJsonException e) {
+            getJolokiaContext().error(String.format("Unexpected error when serializing %s", pValue.getClass().getSimpleName()), e);
+            return null;
+        } catch (JsonIOException e) {
+            getJolokiaContext().error(String.format("Unexpected error when serializing %s", pValue.getClass().getSimpleName()), e);
+            return null;
+        }
+    }
+
+    public Object deserialize(String pExpectedClassName, Object pValue) {
+        if (pValue == null) {
+            return null;
+        }
+        Class<?> expectedClass = ClassUtil.classForName(pExpectedClassName);
+        if (expectedClass == null) {
+            return null;
+        }
+        if (pValue instanceof String) {
+            if (((String) pValue).isEmpty() && expectedClass == String.class) {
+                return pValue;
+            }
+        }
+        try {
+            Gson gson = getGson();
+            return gson.fromJson(toJSON(pValue).toJSONString(), expectedClass);
+        } catch (JsonSyntaxException e) {
+            getJolokiaContext().error(String.format("Unexpected error when deserializing %s", pValue), e);
+            return null;
+        }
+    }
+
+    private Gson getGson() {
+        return new GsonBuilder()
+                .serializeNulls()
+                .serializeSpecialFloatingPointValues()
+                .create();
+    }
+
+    public WriteRequestValues setInnerValue(Object pOuterObject, Object pNewValue, List<String> pPathParts) throws AttributeNotFoundException, IllegalAccessException, InvocationTargetException {
+        if (pOuterObject == null) {
+            return null;
+        }
+        String originalJson = ((JSONAware) serialize(pOuterObject, Collections.<String>emptyList(), null)).toJSONString();
+        JsonElement newValue = (JsonElement) deserialize(JsonElement.class.getName(), pNewValue);
+        JSONAware originalValue = (JSONAware) serialize(pOuterObject, pPathParts, null);
+
+        String updatedJson;
+        if (originalValue == null) {
+            String lastPathElement = pPathParts.remove(pPathParts.size() - 1);
+            String path = buildPath(pPathParts);
+            updatedJson = JsonPath.parse(originalJson, jsonPathconfiguration).put(path, lastPathElement, newValue).jsonString();
+        } else {
+            String path = buildPath(pPathParts);
+            updatedJson = JsonPath.parse(originalJson, jsonPathconfiguration).set(path, newValue).jsonString();
+        }
+        return new WriteRequestValues(deserialize(pOuterObject.getClass().getName(), updatedJson), originalValue);
+    }
```

Comment:
Hello, 

Sorry for the delay.

I feel like in theory the second option is better as this does not have to do a lot with serializer purpose. It may result in an important code impact and dependencies (aka serialization) will be a problem as you stated above. There's no quick win here from code perspective. 

This is why in practice I would vote for the first option, mainly for its simplicity. 

Kind regards, 

Vladimir